matplotcheck/raster.py

```diff
@@ -161,21 +162,54 @@ def assert_legend_accuracy_classified_image(
             im_data_labels, im_expected_labels
         ), "Incorrect legend to data relation"
 
-        ### IMAGE TESTS/HELPER FUNCTIONS ###
+    def _image_shape_correction(
```

Comment:
@nkorinek doesn't this need self so we can also call it on the plot object itself? i need to look at this a bit more closely but it seems like all methods should have a `self` call.

Reply:
I had it set up like that originally, but Codacy flagged it as bad form. It noted that since the method never called on the `self` parameter it should just be made into a function instead.

Reply:
ahhhh ok interesting. can you still call the method on the object without self however? it seems like something is off here but i could be wrong! i can grab the code in the next day or two and play with it more. it seems like we want to be able to just use the helper if need be. it's within the raster tester object is the part that is throwing me off here.  it's not actually a function it's a method i think?? can you call it outside of the plottester object?

Reply:
What does this function expect? does it expect a numpy array? are we assuming the user has NOT plotted the data yet? 

i'm asking because it's hard to know whether a raster has been transposed or not. maybe we can have a quick chat tomorrow on slack about this. i just want to think it through as i know what you are doing... but i also wondered if you can just create the plot in the answers and create a MPC object from the expected plot. Grab the image from that and use in the tests. Do you know what i mean? we can chat more tomorrow because it is also possible i'm over thinking things :)

Reply:
so to pseudo code this out:
```
### Begin solution
# PLOT data here 
# grab MPC object from expected plot
# USE MPC object to extract the expected image which will already be transposed and such then give that to the tests below
### end solution
```

Reply:
@lwasser I think I see the confusion here, but we should definitely chat about this tomorrow. This is not modifying the MPC object. It's modifying the data put into the MPC object to compare against. It doesn't have to be stored in the `RasterTester` object, since it is just a function and not a method. It doesn't need to be a method, at least the way I'm doing it. The way you outlined in your psuedo code would work as well, we can talk about which way we want to go with!